tests/strategies/test_ddp_hydra_support.py

```diff
@@ -0,0 +1,157 @@
+import logging
+import os
+import subprocess
+import sys
+import tempfile
+from pathlib import Path
+
+import pytest
+
+from pytorch_lightning.utilities.imports import _HYDRA_AVAILABLE
+from tests.helpers.runif import RunIf
+
+if _HYDRA_AVAILABLE:
+    from omegaconf import OmegaConf
+
+
+# fixture to run in a clean temporary directory
+@pytest.fixture()
+def cleandir():
+    """Run function in a temporary directory."""
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        old_dir = os.getcwd()  # get current working directory (cwd)
+        os.chdir(tmpdirname)  # change cwd to the temp-directory
+        yield tmpdirname  # yields control to the test to be run
+        os.chdir(old_dir)
+        logging.shutdown()
+
+
+# function to run a command line argument
+def run_process(cmd):
+    try:
+        process = subprocess.Popen(
+            args=cmd,
+            shell=False,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        stdout, stderr = process.communicate()
+        if process.returncode != 0:
+            sys.stderr.write(f"Subprocess error:\n{stderr}\n")
+            raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)
+        return stdout, stderr
+    except Exception as e:
+        cmd = " ".join(cmd)
+        sys.stderr.write(f"Error executing:\n{cmd}\n")
+        raise e
+
+
+# Script to run from command line
+script = """
+import os
+
+import hydra
+import torch
+from torch import distributed as dist
+
+from pytorch_lightning import Trainer
+from pytorch_lightning.strategies import DDPStrategy
+from tests.helpers.boring_model import BoringModel
+
+
+class BoringModelGPU(BoringModel):
+    def on_train_start(self) -> None:
+        # make sure that the model is on GPU when training
+        assert self.device == torch.device(f"cuda:{self.trainer.strategy.local_rank}")
+        self.start_cuda_memory = torch.cuda.memory_allocated()
+
+
+@hydra.main()
+def task_fn(cfg):
+    trainer = Trainer(gpus=cfg.gpus, strategy=cfg.strategy, fast_dev_run=True)
+    model = BoringModelGPU()
+    trainer.fit(model)
+
+    # Need to do this in addition to Lightning shutting down the
+    # distributed processes in order to run a multirun loop with hydra
+    if dist.is_initialized():
```

Comment:
Can we detect the user is using multirun with Hydra and do this automatically within the DDP Plugin ?

Reply:
I'll look into it, but I think you cannot.

Reply:
I cannot find any mechanism to detect multirun.  Each job is a standard run job and only the sweeper is aware of the multirun.  Maybe PL could have an option to fully shutdown distributed and clean the environment?

Reply:
@jgbos  could you inspect `HydraConf.sweep`? I think `HydraConf.sweep.dir is not MISSING` might be a reliable signal that multirun is happening

Reply:
Unfortunately, as far as I can tell both `run.dir` and `sweep.dir` are set no matter the run mode.

Reply:
Oh, I may have found it.  We can check `OmegaConf.is_missing(hydra_cfg.job, "num")`.

Reply:
@tchaton is the `teardown` the appropriate place to do this?